### PR TITLE
fix:error when adding the ollama embedding model

### LINKED ISSUE #8209: Ollama reports an error when adding Text Embedding.
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.8.0

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

An error occurred during credentials validation: Session.request() got an unexpected keyword argument 'options'

### ✔️ Expected Behavior

_No response_

### ❌ Actual Behavior

_No response_

### PR DESCRIPTION
**Pull request by Howe829**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [ ] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. Close issue syntax: `Fixes #<issue number>`, see [documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) for more details.

Fixes #8209 

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

**Comment by AAEE86** [comment]
You can Fixes #8209 in Description.

**Review by laipz8200 — DISMISSED** [review]

File: api/core/model_runtime/model_providers/ollama/text_embedding/text_embedding.py [hunk 1]
```diff
@@ -80,11 +80,14 @@ def _invoke(
         payload = {
             "input": inputs,
             "model": model,
+            "options": {
+                "use_mmap": True
```
Comment (crazywoola) [inline api/core/model_runtime/model_providers/ollama/text_embedding/text_embedding.py]:
```suggestion
                "use_mmap": "true"
```

**Review by crazywoola — COMMENTED** [review]
It should be "true", otherwise it will report 500 error.

**Review by crazywoola — APPROVED** [review]

**Comment by crazywoola** [comment]
Maybe fix https://github.com/langgenius/dify/issues/6493

**Comment by Howe829** [comment]
> It should be "true", otherwise it will report 500 error.

Hi, actually I think "true" will cause 500 error
image

**Comment by AAEE86** [comment]
Looks like it should be True.

You can create a new PR.

**Comment by kksmi** [comment]
I changed that place in [api/core/model_runtime/model_providers/ollama/text_embedding/text_embedding.py] to True, but it still reports an error when adding the embedded model.

**Comment by Howe829** [comment]
> I changed that place in [api/core/model_runtime/model_providers/ollama/text_embedding/text_embedding.py] to True, but it still reports an error when adding the embedded model.

Could you please paste the error here? I think you can check whether you can connect to the Ollama API server or if the Base URL of the Ollama server is correct
image
image

**Comment by kksmi** [comment]
> > I changed that place in [api/core/model_runtime/model_providers/ollama/text_embedding/text_embedding.py] to True, but it still reports an error when adding the embedded model.
> 
> Could you please paste the error here? I think you can check whether you can connect to the Ollama API server or if the Base URL of the Ollama server is correct image image

image
The URL and name are correct and it can be added as LLM.

**Comment by AAEE86** [comment]
> > > I changed that place in [api/core/model_runtime/model_providers/ollama/text_embedding/text_embedding.py] to True, but it still reports an error when adding the embedded model.
> > 
> > 
> > Could you please paste the error here? I think you can check whether you can connect to the Ollama API server or if the Base URL of the Ollama server is correct image image
> 
> image The URL and name are correct and it can be added as LLM.

 ```
payload = {"input": inputs, "model": model, "options": {"use_mmap": "true"}}
# Make the request to the Ollama API
response = requests.post(endpoint_url, headers=headers, data=json.dumps(payload), timeout=(10, 300))
```
Are you sure it was modified according to the above code?

**Comment by Howe829** [comment]
> > > I changed that place in [api/core/model_runtime/model_providers/ollama/text_embedding/text_embedding.py] to True, but it still reports an error when adding the embedded model.
> > 
> > 
> > Could you please paste the error here? I think you can check whether you can connect to the Ollama API server or if the Base URL of the Ollama server is correct image image
> 
> image The URL and name are correct and it can be added as LLM.

Could you show me the code you changed? I guess you are running it with docker and the error you showed above is already fixed by this pr. The True or "true" problem is an other issue.

**Comment by AAEE86** [comment]
我觉得可以尝试将 "options": {"use_mmap": True} 删除，我不太清楚它在这里起什么作用？

**Comment by kksmi** [comment]
> 我觉得可以尝试将 "options": {"use_mmap": True} 删除，我不太清楚它在这里起什么作用？

去掉也运行不了好像

**Comment by crazywoola** [comment]
> 我觉得可以尝试将 "options": {"use_mmap": True} 删除，我不太清楚它在这里起什么作用？

Try to fix this one. https://github.com/langgenius/dify/issues/8128

**Comment by kksmi** [comment]
> 
> 
> i have change the file ,But when I uploaded the knowledge base，Still the following error occurs： 
> 
> 

你去掉之后还会提示错误吗，我还是会，有部分文件不会，但是感觉没有索引成功

**Comment by Howe829** [comment]
> > 
> > i have change the file ,But when I uploaded the knowledge base，Still the following error occurs： 
> > 
> 
> 你去掉之后还会提示错误吗，我还是会，有部分文件不会，但是感觉没有索引成功

Did you restart your celery worker?

**Comment by liuxiaohao-xn** [comment]
> > > 我已经更改了文件，但是当我上传知识库时，仍然出现以下错误：
> > > 
> > 
> > 
> > 你去掉之后还会提示错误吗，我还是会，有部分文件不会，但是感觉没有索引成功
> 
> 你重新启动你的芹菜工人了吗？我可以成功索引并进行检索 image

我全部镜像都手动重启了，但是在创建知识库的时候还是会报这样的错误
image

**Comment by kksmi** [comment]
> > > > 我已经更改了文件，但是当我上传知识库时，仍然出现以下错误：
> > > > 
> > > 
> > > 
> > > 你去掉之后还会提示错误吗，我还是会，有部分文件不会，但是感觉没有索引成功
> > 
> > 
> > 你重新启动你的芹菜工人了吗？我可以成功索引并进行检索 image
> 
> 我全部镜像都手动重启了，但是在创建知识库的时候还是会报这样的错误 image

手动重启之后，如果你是windows，需要修改worker和api下该文件，去掉该文件的的options就可以跑了

**Comment by kksmi** [comment]
> > 我觉得可以尝试将 "options": {"use_mmap": True} 删除，我不太清楚它在这里起什么作用？
> 
> Try to fix this one. #8128

It turns out that removing the option is the only way to run it, the error may be related to his version of ollama, there is no need to include options, I hope to remove options in the next api image upgrade, it's not convenient for windows users, thank you very much.

**Comment by Howe829** [comment]
> > > 我觉得可以尝试将 "options": {"use_mmap": True} 删除，我不太清楚它在这里起什么作用？
> > 
> > 
> > Try to fix this one. #8128
> 
> It turns out that removing the option is the only way to run it, the error may be related to his version of ollama, there is no need to include options, I hope to remove options in the next api image upgrade, it's not convenient for windows users, thank you very much.

I think you can check the Ollama logs to see if there are any errors. I highly recommend upgrading Ollama to the latest version.

**Comment by verigle** [comment]
添加文档的时候依然会在80%左右的时候出现 ollama embedding 500 错误

**Pull request merged**